src/Utilities/Compiler/FileNameUtilities.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Analyzer.Utilities
+{
+    /// <summary>
+    /// Copied from Roslyn.Utilities.FileNameUtilities.
```

Comment:
Can we file a tracking issue to remove this code once https://github.com/dotnet/roslyn/issues/39758 is implemented? Also please add a TODO in this comment mentioning the github URI of the new issue filed. Thanks.

Reply:
Actually, can we just make this a private nested type of GeneratedCodeUtilities, so no one else ends up using these helpers? You can move the above comment and TODO directly to GeneratedCodeUtilities type.